Lib/test/test_asyncio/test_unix_events.py

```diff
@@ -504,12 +504,9 @@ def prepare(self):
             lambda: proto, sock=srv_sock))
         self.run_loop(self.loop.sock_connect(sock, (support.HOST, port)))
 
-        def cleanup():
-            if proto.transport is not None:
-                # can be None if the task was cancelled before
-                # connection_made callback
-                proto.transport.close()
-                self.run_loop(proto.wait_closed())
+        def cleanup(transport=proto.transport):
```

Comment:
Looks not robust and cumbersome.
Better to not pass a transport but 
1. Create a future in MyProto
2. Set the future to some value (`None` for example) in `proto.connection_made()`
3. Wait for the future after `loop.sock_connect()` call

By this `prepare()` always return a connected server protocol, the change should eliminate race condition

Reply:
Done in 1773bbd. Thanks for the suggestion! :)